### PR TITLE
Implement ProfileRouter for deep account / lock links (3333)

### LINKED ISSUE #3333: Add "View lock details" link in the unavailable accounts due to locking tooltip
## Context
Core issue is in this ticket:
- https://github.com/Joystream/pioneer/issues/3238

## Scope

- Add anchors to account lock details page so that user can be directed to an expanded lock + screen focus. 
- Add "View lock details>" to the tooltip used for unavailable accounts due to locks: 

Below the copy and above "Learn more about locks>"

### PR DESCRIPTION
**Pull request by traumschule**
Fixes #3333

This expands existing `ProfileRoutes` and adds `ProfileModule` to `App`.

Following routes have been implemented:
- /profile/accounts/:accountId
- /profile/accounts/:accountId/locks/:lockId

Notes
- No UI work to scroll, focus or expand has been done and is OOS.
- No address lookup if membership handle is passed.
- There could be a smarter way to pass `address` to tooltips.
- The `Review all account locks` tooltip link shows correctly but doesn't route. Needs a new routing tooltip component.
- The tooltip scrolls around (known bug), once the accounts dropdown jumped to the right and looked broken.

## Preview

The link target displays correct account ID and console confirms that params are interpreted on MyAccounts.

account-link

profile-locks

### CODE REVIEW

File: packages/ui/src/accounts/components/SelectAccount/AccountLockTooltip.tsx [hunk 1]
```diff
@@ -13,13 +14,14 @@ import { SwitchMemberModalCall } from '@/memberships/modals/SwitchMemberModal'
 import { OptionLock } from '../../types'
 
 interface Props {
+  address: string
   locks?: OptionLock[]
   children: React.ReactNode
-  boundaryClassName?: string
```
[the remark this answers is not in the document]
Comment (traumschule) [reply packages/ui/src/accounts/components/SelectAccount/AccountLockTooltip.tsx:18]: Nothing in the codebase uses it.

File: packages/ui/src/accounts/components/AccountLocks.tsx [hunk 2]
```diff
@@ -52,6 +52,7 @@ const locksMap: Record<LockType, ReactElement> = {
 export const lockIcon = (type: LockType) => locksMap[type] ?? <QuestionIcon />
 
 export interface AccountLocksProps {
+  address: string
```
[the remark this answers is not in the document]
Comment (traumschule) [reply packages/ui/src/accounts/components/AccountLocks.tsx:55]: Fair point, not sure anymore :)

File: packages/ui/src/accounts/components/SelectAccount/AccountLockTooltip.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (WRadoslaw) [reply packages/ui/src/accounts/components/SelectAccount/AccountLockTooltip.tsx:18]: Tooltip in this very file.  It is used to flip the tooltip when it detaches from the dropdown

` <Tooltip boundaryClassName={boundaryClassName} tooltipText={tooltipTexts} maxWidth>`

[the remark this answers is not in the document]
Comment (traumschule) [reply packages/ui/src/accounts/components/SelectAccount/AccountLockTooltip.tsx:18]: ok, i trust you on this. guess i need a vacation.

File: packages/ui/src/accounts/components/AccountLocks.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (traumschule) [reply packages/ui/src/accounts/components/AccountLocks.tsx]: This is why, just realized. Expecting it to get used later on.
```
\| ERROR in src/accounts/components/AccountItem/AccountItem.tsx:45:25
--
08:59:13.528 | \| TS2322: Type '{ address: string; locks: BalanceLock[] \| undefined; }' is not assignable to type 'IntrinsicAttributes & AccountLocksProps'.
08:59:13.528 | \|   Property 'address' does not exist on type 'IntrinsicAttributes & AccountLocksProps'.
08:59:13.528 | \|     43 \|         <ValueAndLocks align={balance?.locked && 'end'}>
08:59:13.528 | \|     44 \|           <TokenValue value={balance?.locked} isLoading={!isDefined(balance?.locked)} />
08:59:13.528 | \|   > 45 \|           <AccountLocks address={address} locks={balance?.locks} />
08:59:13.528 | \|        \|                         ^^^^^^^
08:59:13.528 | \|     46 \|         </ValueAndLocks>
08:59:13.528 | \|     47 \|         <TokenValue
08:59:13.528 | \|     48 \|           value={balance?.recoverable?.add(balance?.vestedClaimable ?? BN_ZERO)}
08:59:13.528 | \|
08:59:13.528 | \| ERROR in src/accounts/components/SelectAccount/OptionAccount.tsx:30:25
08:59:13.528 | \| TS2322: Type '{ address: string; locks: BalanceLock[] \| undefined; }' is not assignable to type 'IntrinsicAttributes & AccountLocksProps'.
08:59:13.528 | \|   Property 'address' does not exist on type 'IntrinsicAttributes & AccountLocksProps'.
08:59:13.529 | \|     28 \|         <InfoValueWithLocks>
08:59:13.529 | \|     29 \|           <Value value={balance} locked={isLocked} />
08:59:13.529 | \|   > 30 \|           <AccountLocks address={option.address} locks={balances?.locks} />
08:59:13.530 | \|        \|                         ^^^^^^^
08:59:13.530 | \|     31 \|         </InfoValueWithLocks>
08:59:13.530 | \|     32 \|       </BalanceInfoInRow>
08:59:13.530 | \|     33 \|     </>
08:59:13.530 | \|
08:59:13.530 | \| ERROR in src/memberships/components/MemberListItem/Member.tsx:65:25
08:59:13.531 | \| TS2322: Type '{ address: string; locks: BalanceLock[] \| undefined; }' is not assignable to type 'IntrinsicAttributes & AccountLocksProps'.
08:59:13.531 | \|   Property 'address' does not exist on type 'IntrinsicAttributes & AccountLocksProps'.
08:59:13.531 | \|     63 \|         <MemberColumn>
08:59:13.531 | \|     64 \|           <TokenValue value={balance?.locked} />
08:59:13.531 | \|   > 65 \|           <AccountLocks address={member.rootAccount} locks={balance?.locks} />
08:59:13.531 | \|        \|                         ^^^^^^^
08:59:13.531 | \|     66 \|         </MemberColumn>
08:59:13.531 | \|     67 \|       </MemberItemWrap>
08:59:13.532 | \|     68 \|       {searchFilter && !EMPTY_SEARCH_FILTERS.includes(searchFilter) && (
```

**Comment by traumschule** [comment]
> While I understand passing address to the tooltip and routing, but why pass it to the locks?

Summarizing the description `accountId` shall be used to route to the account (on `My Accounts`) from inside the lock tooltip to `Review all account locks`.

File: packages/ui/src/accounts/components/SelectAccount/OptionListAccount.tsx [hunk 3]
```diff
@@ -25,7 +25,7 @@ export const OptionListAccount = React.memo(({ options, onChange, className, isF
         </Option>
       ))}
       {lockedAccounts.map((option) => (
-        <AccountLockTooltip boundaryClassName={className} key={option.address} locks={option.optionLocks}>
```
Comment (oleksanderkorn) [inline packages/ui/src/accounts/components/SelectAccount/OptionListAccount.tsx:28]: should this `boundaryClassName={className}` be removed?

**Review by oleksanderkorn — APPROVED** [review]

File: packages/ui/src/memberships/components/MemberListItem/Member.tsx [hunk 4]
```diff
@@ -26,7 +26,7 @@ interface MemberListItemProps {
 const EMPTY_SEARCH_FILTERS: MemberSearchFilter[] = ['Membership', 'Membership_ID', 'Account_Address']
 
 export const MemberListItem = ({ member, searchFilter }: MemberListItemProps) => {
-  const balance = useBalance(member.controllerAccount)
+  const balance = useBalance(member.controllerAccount) // better root?
```
Comment (thesan) [inline packages/ui/src/memberships/components/MemberListItem/Member.tsx:29]: This is out of scope here. You could open an issue about this.
```suggestion
  const balance = useBalance(member.controllerAccount)
```

Although the way I see it,  controller account makes more sense to me here. Because most transaction will target the user controller account. OTOH the root account is more of a security (best kept offline) to recover memberships.

File: packages/ui/src/memberships/components/MemberListItem/Member.tsx [hunk 5]
```diff
@@ -62,7 +62,7 @@ export const MemberListItem = ({ member, searchFilter }: MemberListItemProps) =>
 
         <MemberColumn>
           <TokenValue value={balance?.locked} />
-          <AccountLocks locks={balance?.locks} />
+          <AccountLocks address={member.rootAccount} locks={balance?.locks} />
```
Comment (thesan) [inline packages/ui/src/memberships/components/MemberListItem/Member.tsx:65]:
```suggestion
          <AccountLocks locks={balance?.locks} />
```

**Comment by traumschule** [comment]
```
ERROR in src/accounts/components/SelectAccount/AccountLockTooltip.tsx:102:23
--
17:43:34.456 | \| TS2304: Cannot find name 'generatePath'.
17:43:34.456 | \|     100 \|       tooltipText={tooltipTexts}
17:43:34.456 | \|     101 \|       tooltipLinkText="Review all account locks"
17:43:34.456 | \|   > 102 \|       tooltipLinkURL={generatePath(ProfileRoutes.account { accountId: address })}
17:43:34.456 | \|         \|                       ^^^^^^^^^^^^
```
